Add HWI example in docs

danielabrozzoni opened the pull request:
### Description


### Notes to the reviewers


### Changelog notice

- bdk re-exports the `hwi` create when the feature `hardware-signer` is on
- Add `examples/hardware_signer.rs`

### Checklists

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

afilini commented on src/wallet/hardwaresigner.rs:
```diff
@@ -12,6 +12,62 @@
+//!
+//! fn main() -> Result<(), bdk::Error> {
+//!     // Listing all the available hardware wallet devices...
+//!     let devices = HWIClient::enumerate().unwrap();
```
No need to unwrap if you return a result

afilini replied:
Same for all the other unwraps down below

afilini commented on src/wallet/hardwaresigner.rs:
```diff
@@ -12,6 +12,62 @@
+//!     let client = HWIClient::get_client(
+//!         devices.first().expect(
+//!             "No devices found. Either plug in a hardware wallet, or start a simulator.",
+//!         ),
```
Consider assigning `devices.first()` to a variable because you use it down below as well

afilini commented on src/wallet/hardwaresigner.rs:
```diff
@@ -12,6 +12,62 @@
+//!         HWISigner::from_device(devices.first().unwrap(), HWIChain::Test).unwrap();
+//!     let mut wallet = Wallet::new(
+//!         &descriptors.internal[0],
+//!         Some(&descriptors.receive[0]),
```
Those two should be swapped around I think

danielabrozzoni commented:
Moving this to the 0.24 milestone, as I need to tag the newest hwi first (I need https://github.com/bitcoindevkit/rust-hwi/commit/ab07d1379c43336393cf62cbc72e63a22f7c589c)

danielabrozzoni commented:
Tagging hwi will take some time, putting this in draft

danielabrozzoni committed:
Re-export HWI if the hardware-signer feature is set

danielabrozzoni committed:
Bump HWI to 0.2.3

danielabrozzoni committed:
Update the hardwaresigner module documentation

Add a little example on how to use the HWISigner, slightly improve
the module description

danielabrozzoni committed:
Add the hardware_signer example

danielabrozzoni commented:
Sorry if this PR was in a rather crappy state @afilini. It should definitely be better now! :)

afilini reviewed (approved):
ACK 1437e1ecfe663b819156d98c5e1975fb357a763f
